packages/contracts-bedrock/src/universal/StandardBridgeExtension.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { StandardBridge } from "./StandardBridge.sol";
+
+abstract contract StandardBridgeExtension is StandardBridge {
+    address public immutable LOCAL_TOKEN;
+    address public immutable REMOTE_TOKEN;
+
+    constructor(
+        address payable _messenger,
+        address payable _otherBridge,
+        address _localToken,
+        address _remoteToken
+    ) StandardBridge(_messenger, _otherBridge) {
+        LOCAL_TOKEN = _localToken;
+        REMOTE_TOKEN = _remoteToken;
+    }
+
+    modifier onlyERC20(address _localToken, address remoteToken) {
```

Comment:
This is very particular for only a single token per bridge which may be good for SNX and DAI but unclear to me if other projects would want something else. If this is meant just for them then i understand that